js/frame.js

```diff
@@ -0,0 +1,76 @@
+'use strict';
+
+function Frame() {
+  this.MAX_PINS = 10;
+  this.score = 0;
+  this.rollCount = 0;
+  this.bonusRollsNeeded = 0;
+  this.completion = false;
+};
+
+Frame.prototype.roll = function(pins) {
+  if(pins + this.getScore() > this.MAX_PINS || pins < 0) { throw('Received wrong number of pins') };
+  this._rollCountUp();
+  this._increaseScore(pins);
+  this._setBonusIfEligible();
+  this._completeFrameIfEligible();
+};
+
+Frame.prototype.needsBonus = function() {
+  return this.bonusRollsNeeded > 0;
+};
+
+Frame.prototype.addBonus = function(pins) {
+  this._increaseScore(pins);
+  this.bonusRollsNeeded--;
+};
+
+Frame.prototype.getScore = function() {
+  return this.score;
+};
+
+Frame.prototype.isComplete = function() {
+  return this.completion;
```

Comment:
Looks like you could remove some lines and call `_completeFrameIfEligible()` here?